### PR TITLE
Add support for logging to console and files

### PR DESCRIPTION
**Pull request by LennardF1989**
Adds a logger based on [Winston](https://github.com/winstonjs/winston):
- Creates a new log every time Peacock is started (`frequency` is required, since the default value - `custom` - doesn't work as well).
- Keep a maximum amount of files (14 days right now, but can be tweaked)
- Files will have a JSON object per line with information: `{"timestamp":"02:50:58:732","level":"Info","message":"Booting Peacock internal services - this may take a moment."}`
- When debug mode is on, will also log `debug` and `trace` level, otherwise `info` and higer.
- Console still has colors and usual output

### CODE REVIEW

File: components/loggingInterop.ts [hunk 1]
```diff
@@ -19,6 +19,44 @@
...
 import picocolors from "picocolors"
+import winston from "winston"
+import "winston-daily-rotate-file"
+
+const isDebug = ["*", "true", "peacock", "yes"].includes(
+    process.env.DEBUG || "false",
+)
+
+const fileTransport = new winston.transports.DailyRotateFile({
+    filename: "logs/application-%DATE%.json",
```
Comment (LennardF1989) [inline components/loggingInterop.ts:30]: This should probably just be "peacock" instead of application.

File: components/loggingInterop.ts [hunk 2]
```diff
@@ -19,6 +19,44 @@
...
+    datePattern: "YYYYMMDDHHmmss",
+    frequency: "1d",
+    maxFiles: "14d",
```
Comment (LennardF1989) [inline components/loggingInterop.ts:33]: Make this configurable, eg. by means of an environment variable.

**Review by LennardF1989 — COMMENTED** [review]
Notes.

**Comment by moonysolari** [comment]
When debug mode is off, can it be such that messages of all categories (including debug and trace level) will appear in files, while only those of info and higher will appear in the console?

**Comment by LennardF1989** [comment]
> When debug mode is off, can it be such that messages of all categories (including debug and trace level) will appear in files, while only those of info and higher will appear in the console?

Yes, that's possible.

File: components/loggingInterop.ts [hunk 3]
```diff
@@ -19,6 +19,44 @@
...
+    }),
+})
+
+const logger = winston.createLogger({
+    levels: {
+        error: 0,
+        warn: 1,
+        info: 2,
+        debug: 3,
+        trace: 4,
```
Comment (RDIL) [inline components/loggingInterop.ts:55]: Can we use the LogLevel enum here?

Comment (LennardF1989) [reply components/loggingInterop.ts:55]: Check the next commit - this is the best we can do with Winston.

**Comment by LennardF1989** [comment]
> I want to do this in a way that works great and still looks pretty.

Well, got news, because the console output looks exactly the same as it was before!

**Comment by LennardF1989** [comment]
This is now a "one logger to rule them all" solution. Preferable, you want a logger instance per service, which all tunnel down into the same place eventually.

The easiest would be adding an additional enum to the log with a list of category we can filter on, it's then relatively easy to turn them on/off as bitflags.

**Comment by LennardF1989** [comment]
Latest commit has all you requested @RDIL and also integraties support for #44.

**Comment by LennardF1989** [comment]
@moonysolari Yes, it does. Default log level for the file is TRACE and unless explicitly turned off, it logs everything even if the console doesn't.

**Review by RDIL — APPROVED** [review]

**Pull request merged**